docs/docs/icechunk-python/version-control.md

```diff
@@ -103,141 +98,138 @@ Now that we've created a new snapshot, we can time-travel back to the previous s
 
     It's important to note that because the `zarr Store` is read-only, we need to pass `mode="r"` to the `zarr.open_group` function.
 
-```python
-session = repo.readonly_session(snapshot_id="BSHY7B1AGAPWQC14Q18G")
+```python exec="on" session="version" source="material-block" result="code"
+session = repo.readonly_session(snapshot_id=list(repo.ancestry(branch="main"))[1].id)
 root = zarr.open_group(session.store, mode="r")
-root.attrs["foo"]
-
-# 'bar'
+print(root.attrs["foo"])
 ```
 
 ## Branches
 
 If we want to modify the data from a previous snapshot, we can create a new branch from that snapshot with [`create_branch`](../reference/#icechunk.Repository.create_branch).
 
-```python
+```python exec="on" session="version" source="material-block"
+main_branch_snapshot_id = repo.lookup_branch("main")
 repo.create_branch("dev", snapshot_id=main_branch_snapshot_id)
 ```
 
 We can now create a new writable `Session` from the `dev` branch and modify the data.
 
-```python
+```python exec="on" session="version" source="material-block" result="code"
 session = repo.writable_session("dev")
 root = zarr.group(session.store)
 root.attrs["foo"] = "balogna"
-session.commit(message="Update foo attribute on root group")
-
-# 'H1M3R93ZW19MYKCYASH0'
+print(session.commit(message="Update foo attribute on root group"))
 ```
 
 We can also create a new branch from the tip of the `main` branch if we want to modify our current working branch without modifying the `main` branch.
 
-```python
-main_branch_snapshot_id = repo.lookup_branch("main")
+```python exec="on" session="version" source="material-block" result="code"
 repo.create_branch("feature", snapshot_id=main_branch_snapshot_id)
 
 session = repo.writable_session("feature")
 root = zarr.group(session.store)
 root.attrs["foo"] = "cherry"
-session.commit(message="Update foo attribute on root group")
-
-# 'S3QY2RDQQTRYFGJDTB6G'
+print(session.commit(message="Update foo attribute on root group"))
 ```
 
-With these branches created, the hierarchy of the repository now looks like below.
+With these branches created, the hierarchy of the repository now looks like below. 
 
-```mermaid
+```python exec="on" result="mermaid" session="version"
+main_commits = [s.id[:6] for s in list(repo.ancestry(branch='main'))]
```

Comment:
We've wanted a method to do this for quite long!

Reply:
You mean creating the mermaid diagram from python? it would be amazing if it could just auto generate the tree for you in a notebook. Would just have to bundle `mermaid.js`